core/src/main/java/org/apache/druid/utils/CollectionUtils.java

```diff
@@ -83,6 +86,17 @@ public int size()
     return result;
   }
 
+  /**
+   * Returns a transformed map from the given input map where the key is modified based on the given keyMapper
+   * function.
+   */
+  public static <K, V, K2> Map<K2, V> mapKeys(Map<K, V> map, Function<K, K2> keyMapper)
+  {
+    final Map<K2, V> result = Maps.newHashMapWithExpectedSize(map.size());
+    map.forEach((k, v) -> result.put(keyMapper.apply(k), v));
```

Comment:
I think it's better if this method fails with IllegalStateException on keys collisions if the key mapper is not a bijection, similar to [`ImmutableMap.Builder`](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html). This should also be reflected in the method's javadoc.